deepchem/models/dft/nnxc.py

```diff
@@ -0,0 +1,152 @@
+from abc import abstractproperty, abstractmethod
+from typing import Union
+try:
+    from dqc.xc.base_xc import BaseXC
+    from dqc.utils.datastruct import ValGrad, SpinParam
+    from dqc.api.getxc import get_xc
+    import torch
+except ModuleNotFoundError:
+    raise ModuleNotFoundError("This layer requires dqc and torch")
+
+
+class BaseNNXC(BaseXC, torch.nn.Module):
```

Comment:
All these layers should be added to the `docs/` folder so they render in the layer documentation